pkg/k8shandler/deployment.go

```diff
@@ -365,25 +371,33 @@ func (node *deploymentNode) executeUpdate() error {
 }
 
 func (node *deploymentNode) progressNodeChanges() error {
-	if node.isChanged() || !node.podSpecMatches() {
-		if err := node.executeUpdate(); err != nil {
-			return err
-		}
+	if !node.isChanged() || node.podSpecMatches() {
```

Comment:
shouldn't this be `&&` since you negated this?

Reply:
I don't believe so since they're both negated. I'm reading "if the node hasn't changed or the specs match then return nil"

Did I misinterpret this?

Reply:
slightly... if the node isn't changed (as in the deployment matches what we built based on the spec) and the pod spec matches this deployment then we're already done. but if the node isn't changed but the pod hasn't been rolled out yet (so it doesn't match the deployment spec) then we need to go through that update function still